### PR TITLE
feat(generator): introduce a generator concept

### PR DESCRIPTION
**Pull request by zlesnr**

### CODE REVIEW

File: templates/clientgo/types.go.tmpl [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package {{.PackageName}}
+
+{{range .Enums}}
+type {{.Name}} string
+{{$typeName := .Name}}
+
+var {{.Name}}Types = struct {
+  {{- range .Values}}
+	/* {{ .Description }}*/
```
Comment (sanderblue) [inline templates/clientgo/types.go.tmpl:9]: Is this comment style a Go-specific thing for displaying documentation on godoc.org? If not, any thoughts on standardizing on `//` for comments and godoc things?

Comment (zlesnr) [reply templates/clientgo/types.go.tmpl:9]: The `/* */` is a multi-line comment for go.  I think we need a method to take the multi-line doc string and convert it to the `//` style.  We have some filtering methods currently, but nothing I saw to transform the string.  Perhaps the `GetDescription()` could do this for us.

File: README.md [hunk 2]
```diff
@@ -12,7 +12,15 @@
...
 
-Generate Golang types from GraphQL schema introspection
+Generate code from GraphQL schema introspection.
+
+## Summary
+
+At a high level, the following workflow is used.
+
+* `tutone fetch` calls the nerdgraph api to introspect the schema.
+* The schema is cached in `schema.json`.  This is information about hte graphql schema
```
Comment (SowmyaJujjuri) [inline README.md:22]: Fix the spelling of word "the"

Comment (zlesnr) [reply README.md]: Updated.

File: templates/clientgo/types.go.tmpl (hunk 1 above)
Comment (zlesnr) [reply templates/clientgo/types.go.tmpl]: This should be resolved for cleaner description handling now.

**Pull request merged**